### DISCUSSION TITLE
How to block a variable on the Volt?

### DISCUSSION THREAD

**Discussion by rafaphp** [phalcon/cphalcon#15527] [Q&A]
I'm putting together a project where third parties can edit the templates, I wish it wasn't possible:

{{ this.any }} {{this.view....}}

I would like to disallow the use of the "this" variable in volt code

Thanks!

**Comment by Izopi4a** [comment]
hm... interesting case. If I am understanding you correctly, perhaps upon saving the edits, you can search for "this." and return an error

**Reply by niden** [reply]
That would be my approach as well. You cannot blacklist `this` because it is used in Volt internally so the best way is to fliter it out in the view snippets submitted by the users. For instance you can search for `this.` and then not allow the save.

**Reply by rafaphp** [reply]
Yes, I can build a regex to search for
"this", the problem is that a simple search is not enough, the ideal would be to block the expression evaluation, the this variable can appear in several cases, concatenated with others and this will be complicated to get with regex. Thought there might be a way to do this built into the Volt, I'm studying how to use this with extensions and events, but the documentation is shallow:

https://docs.phalcon.io/4.0/en/volt#events

Sorry about the language, I'm Brazilian and my English is bad

**Reply by quasipickle** [reply]
For what it's worth, your English is actually quite good.
